[Brandon R. Han] ip

brandonrhan opened the pull request:
### Duke-BH
_Duke-BH provides you with an efficient way to manage your daily tasks_

> Please try to use Duke-BH ---- Brandon Han [[source]](https://www.google.com/webhp?hl=en&sa=X&ved=0ahUKEwjOvIuvoev1AhX27nMBHQwVAvYQPAgI)

Features:
- [x] Managing tasks
- [x] Managing deadlines
- [ ] Reminders (coming soon)

Advantages:
- text-based
- easy to learn
- ~~not bad~~ very good to use

All you need to do:
1. download it from [here.](https://github.com/brandonrhan/ip)
2. run it.
3. add your tasks.
4. let it **manage your tasks** for you 😉

Here is the Main class of the program
`public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}`

Commit:
Add Gradle support

brandonrhan committed:
Level-1

brandonrhan committed:
Level-2

brandonrhan committed:
Level-3

brandonrhan committed:
Level-4

brandonrhan committed:
A-TextUiTesting

brandonrhan committed:
add exceptions

brandonrhan committed:
change in if else statment

brandonrhan committed:
add delete function

brandonrhan committed:
add to lower case for first word

brandonrhan committed:
move run and greet function to class BH

brandonrhan committed:
update xception handling

brandonrhan committed:
add save and load function

brandonrhan committed:
add date and time

brandonrhan committed:
debug load function

brandonrhan committed:
add ui and storage

brandonrhan committed:
add package duke

brandonrhan committed:
add test for event

brandonrhan committed:
add javadoc

brandonrhan committed:
Correct some syntax issue base on the coding standard

brandonrhan committed:
add find function to find task using keyword

declanleeler commented on src/main/java/duke/BH.java:
```diff
@@ -0,0 +1,168 @@
+    /**
+     * Returns number of task in the list
+     *
+     * @return numer of task in the list
```
Not sure I like this repetition in the comments, maybe you could change the @return to 
```suggestion
     * @return size of this.arraylist
```

brandonrhan replied:
Nice catch! I will correct this!

declanleeler commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,101 @@
+    /**
+     * Mark the task as done
+     */
+    void mark() {
```
Should this be changed to adhere to coding standards?
```suggestion
    void setMarked() {
```

brandonrhan replied:
Thank you, I will update this!

declanleeler commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,101 @@
+    /**
+     * Unmark the task as not done
+     */
+    void unMark() {
```
Should this be changed to adhere to coding standards?
```suggestion
    void setUnmark() {
```

declanleeler commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,29 @@
+     * Print the logo and greet user
+     */
+    void greet() {
+
```
Could this dead space be deleted?

declanleeler commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,101 @@
+     * @param word The word wish to check
+     * @return Boolean of whether the word is contained in task
+     */
+    boolean containsWord(String word) {
```
I like that the prefix in this boolean method's naming to signal it is a boolean

declanleeler commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+public class Deadline extends Task{
+
+    /**
+     * Construtor of deadline
```
Is there a typo here?
```suggestion
     * Constructor of deadline
```

declanleeler reviewed (commented):
Great work and it can be seen that a lot of effort has been placed into adhering with code quality. There are minimal violations, and even those are small issues. Overall the code was very easy to follow and has good readability.

kashing555 commented on src/main/java/duke/BH.java:
```diff
@@ -0,0 +1,168 @@
+                    int index = Integer.parseInt(inputArray[1]) - 1;
+                    System.out.println("Oh no! \n" + this.unMark(index));
+                } else if (s1.equals("todo")) {
+                    String task = inputArray[1];
+                    Task newTask = new Todo(task);
+                    this.addToList(newTask);
+                    System.out.println(ui.getLine() + "Task added: " + newTask.toString() + "\n" +
+                            "Now you have " + this.getListSize() + " tasks in the list" + ui.getLine());
+                } else if (s1.equals("deadline")) {
+                    String s = inputArray[1];
+                    String[] arr = s.split("/by");
+                    String task = arr[0];
+                    String time = arr[1];
+                    Task newTask = new Deadline(task, time);
+                    this.addToList(newTask);
+                    System.out.println(ui.getLine() + "Task added:" + newTask.toString() + "\n" +
+                            "Now you have " + this.getListSize() + " tasks in the list" + ui.getLine());
+                } else if (s1.equals("event")) {
+                    String s = inputArray[1];
+                    String[] arr = s.split("/at");
+                    String task = arr[0];
+                    String time = arr[1];
+                    Task newTask = new Event(task, time);
+                    this.addToList(newTask);
+                    System.out.println(ui.getLine() + "Task added:" + newTask.toString() + "\n" +
+                            "Now you have " + this.getListSize() + " tasks in the list" + ui.getLine());
+                } else if (s1.equals("delete")) {
+                    int index = Integer.parseInt(inputArray[1]) - 1;
+                    System.out.println(ui.getLine() + "Okay, I have remove this task:\n" +
+                            this.deleteTask(index) + ui.getLine());
+                } else if (s1.equals("check")) {
+                    LocalDate date = LocalDate.parse(inputArray[1]);
+                    this.checkDate(date);
+                } else if (s1.equals("find")) {
+                    String[] arr = input.split(" ", 2);
+                    String word = arr[1];
+                    this.checkWord(word);
+                } else {
+                    System.out.println("Wrong input, please check again");
+                }
```
Might want to consider a switch case instead of better readability.

brandonrhan replied:
Sure I will try use enum in the next version

kashing555 commented on src/main/java/duke/BH.java:
```diff
@@ -0,0 +1,168 @@
+        return task;
+    }
+
+    private String getList() {
+        String s = "";
+        for (int i = 0; i < this.list.size(); i++) {
+            s = s + (i + 1) + ". " + list.get(i) + "\n";
+        }
+        return s;
+    }
```
Dont really know what String s is.

brandonrhan replied:
Oh thanks I think I forgot to check this, I will update it!

kashing555 commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,36 @@
+    public String toString() {
+        return super.toString() + "(at: " + this.date.format(formatter) + ")";
+    }
+
```
Consider deleting additional lines.

kashing555 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,84 @@
+ * Storage class is used for storing data and loading data
+ */
+public class Storage {
+    private final String filePath;
```
filePath is a constant.
```suggestion
    private final String FILE_PATH;
```

brandonrhan replied:
Thank you for pointing out this!

kashing555 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,101 @@
+ */
+public class Task {
+    boolean isDone;
+    protected final String task;
+    protected final String type;
+    protected final String time;
```
Final constant should be all caps.

kashing555 reviewed (commented):
Might want to consider a switch case instead for better readability. Try to remove additional lines and make sure that your final constant are named in all caps.

channne commented on src/main/java/duke/BH.java:
```diff
@@ -0,0 +1,168 @@
+            while (true) {
+                String input = sc.nextLine();
+                String[] inputArray = input.split(" ", 2);
+                String s1 = inputArray[0].toLowerCase();
```
Perhaps you might want to rename the variable `s1` into something more informative (e.g. command)?

brandonrhan replied:
Okay sure I will update this

channne commented on src/main/java/duke/BH.java:
```diff
@@ -0,0 +1,168 @@
+    }
+
+    /**
+     * continue reading input until a bye is detected
```
Might want to change `continue` to `Continues` (following the Javadoc guidelines)

channne commented on src/main/java/duke/BH.java:
```diff
@@ -0,0 +1,168 @@
+     * if input starts with todo, deadline or event, create a corresponding task and add to list
+     * if input starts with delete, delete the corresponding task
+     * if input starts with check, check all the task on the same date
+     * @throws DukeException if wrong input is detected
```
I think there should be an empty line between the description and the parameter section!

channne commented on src/main/java/duke/BH.java:
```diff
@@ -0,0 +1,168 @@
+        System.out.println(ui.getLine());
+    }
+
+    private void addToList(Task task) {
```
Perhaps it would be helpful for the method names to be more consistent, e.g. `addTask` and `deleteTask`, or `addToList` and `deleteFromList`

channne reviewed (commented):
Overall I think the code was very readable and the coding standards were adhered to! :+1: :smiley:

brandonrhan committed:
Merge branch 'add-gradle-support' of https://github.com/brandonrhan/ip into branch-A-Gradle

brandonrhan committed:
Merge commit 'd8398594b7bc43da5eb865321c5a50cec4b3923d'

brandonrhan committed:
update gradle

brandonrhan committed:
change on gradle

brandonrhan committed:
Change Event constructor to public

brandonrhan committed:
add first 3 part of javaFX tutorial

brandonrhan committed:
add GUI

brandonrhan committed:
Merge branch 'branch-A-Gradle'

brandonrhan committed:
Merge branch 'branch-Level-10'

# Conflicts:
#	build.gradle
#	src/main/java/EventTest.java

brandonrhan committed:
add TaskList class

brandonrhan committed:
add parser

brandonrhan committed:
add Ui.echo function

brandonrhan committed:
add javadoc for new classes

brandonrhan committed:
Add assertion

brandonrhan committed:
Revert "Add assertion"

This reverts commit ad5208413c369d733602a2023b35cdb892f7f0da.

brandonrhan committed:
Add assertions

brandonrhan committed:
Storage class: exist an arrow head structure and repeated code in load function

Different kind of tasks perform common behaviors within the if else statement.

The common behaviors	within the if else statement cause code duplication.

Let's move the common code after the if else statement.

brandonrhan committed:
Update Storage.load()

brandonrhan committed:
Update javadoc comments

brandonrhan committed:
Add Storage.createTaskByType() to simplify Storage.load()

brandonrhan committed:
Parser class: execute user command and return the response

The execute function is using if else statement to determine the commandType. It causes code duplication and forms an arrow head structure.

Let's replace the if else statement by switch case.

Using switch is preferable over if else in this situation because it reduces repeated code and improve readability.

brandonrhan committed:
Parser Class: execute user input and return the response

The execute function includes all the details of action for all kind of Commands. It causes the method to be extremely long.

Let's create a Command class and make all different kind of commands to be subclasses of Command class, and then abstract details of action into the runCommand function of each subclasses.

Using Command class reduce the length of Parser.execute function and improve the readability. SLAP is achieved as well.

brandonrhan committed:
Merge pull request #3 from brandonrhan/A-Assertions

Add assertions

brandonrhan committed:
Merge branch 'master' of https://github.com/brandonrhan/ip into A-CodeQuality

# Conflicts:
#	src/main/java/duke/Parser.java

Keep changes

brandonrhan committed:
Merge pull request #2 from brandonrhan/A-CodeQuality

A code quality

brandonrhan committed:
Merge tag 'A-CodeQuality'

brandonrhan committed:
Merge branch 'master' into A-CodeQuality

brandonrhan committed:
Merge branch 'master' of https://github.com/brandonrhan/ip

brandonrhan committed:
Delete unused variable

brandonrhan committed:
Add tag and untag command

brandonrhan committed:
Merge branch 'C-Tagging'

brandonrhan committed:
Change fxml back to original setting

brandonrhan committed:
Update the exist check

brandonrhan committed:
Merge pull request #5 from brandonrhan/A-CodeQuality

Merge

brandonrhan committed:
Add delay before exit

brandonrhan committed:
Better GUI

brandonrhan committed:
Correct some syntax

brandonrhan committed:
Add Ui.png

brandonrhan committed:
Updates all javadoc comments

brandonrhan committed:
Update README.md

brandonrhan committed:
Update README.md

brandonrhan committed:
Update README.md

brandonrhan committed:
Changes filepath

brandonrhan committed:
Update the load method to handle the case when user does not have the dir

brandonrhan committed:
Merge branch 'master' of https://github.com/brandonrhan/ip

brandonrhan committed:
Update javadocs

brandonrhan committed:
checkstyle
